lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -111,14 +114,12 @@ def verify(credit_card, options = {})
       private
 
       def build_checkout_request(amount, payment, options)
-        raise ArgumentError.new("Payment method #{payment.brand} is not supported, check https://docs.withreach.com/docs/credit-cards#technical-considerations") if PAYMENT_METHOD_MAP[payment.brand.to_sym].blank?
-
         {
           MerchantId: @options[:merchant_id],
           ReferenceId: options[:order_id],
           ConsumerCurrency: options[:currency] || currency(options[:amount]),
           Capture: options[:capture] || false,
-          PaymentMethod: PAYMENT_METHOD_MAP[payment.brand.to_sym],
+          PaymentMethod: PAYMENT_METHOD_MAP.fetch(payment.brand.to_sym, 'unsupported'),
```

Comment:
We shouldn't set a default value here. If a payment method is given, we can map it to the appropriate string value to give back to the gateway, but if we default to `unsupported`, that will end up returned to the user as an error of `unsupported` payment type of `unsupported` which would be confusing.

An additional note, because I didn't notice this before. It looks like the strings we map to are just `capitalized` versions of the payment method? I think we could just use the Ruby built in for `#string.upcase`.

Reply:
Excellent, I think my concern for 1 was that we would be short circuiting the customer value and then getting odd responses back, but sense they return parsing errors, your solution here makes sense. Thanks for the reminder on #2 that any underscore value wouldn't tolerate `.upcase`ing.